sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -1111,46 +1110,33 @@ public virtual async ValueTask DisposeAsync()
 
                     // Sequence the events for publishing.
 
-                    var lastSequence = partitionState.LastPublishedSequenceNumber.Value;
-                    var firstSequence = NextSequence(lastSequence);
-
-                    foreach (var eventData in eventSet)
-                    {
-                        lastSequence = NextSequence(lastSequence);
-                        eventData.PendingPublishSequenceNumber = lastSequence;
-                        eventData.PendingProducerGroupId = partitionState.ProducerGroupId;
-                        eventData.PendingProducerOwnerLevel = partitionState.OwnerLevel;
-                    }
+                    var lastSequence = eventBatch.ApplyBatchSequencing(partitionState.LastPublishedSequenceNumber.Value, partitionState.ProducerGroupId, partitionState.OwnerLevel);
```

Comment:
is `partitionState.LastPublishedSequenceNumber.HasValue` guaranteed to be true here?

Reply:
It is.   The service sets a value for it when the link is opened if the idempotent publishing capability is set.  That happens via the `InitializePartitionStateAsync` call on L1108.   If the service can't initialize the link properties, an exception flow is triggered.